tests/import_test.go

```diff
@@ -89,6 +90,34 @@ var _ = Describe("[rfe_id:1115][crit:high][vendor:cnv-qe@redhat.com][level:compo
 		Entry("fail given a large physical size QCOW2 file", utils.LargePhysicalDiskQcow),
 	)
 
+	DescribeTable(`Should fail with "ImagePullFailed" reason if failed to pull image`, func(url string) {
+		dv := utils.NewDataVolumeWithRegistryImport("failed-to-pull-image", "10Gi", "docker://"+url)
+		if dv.Annotations == nil {
+			dv.Annotations = make(map[string]string)
+		}
+		dv.Annotations[controller.AnnImmediateBinding] = "true"
+
+		var err error
+		dv, err = utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, dv)
+		Expect(err).ToNot(HaveOccurred())
+
+		_, err = utils.WaitForPVC(f.K8sClient, dv.Namespace, dv.Name)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Verify dImagePullFailed running condition")
+		runningCondition := &cdiv1.DataVolumeCondition{
+			Type:    cdiv1.DataVolumeRunning,
+			Status:  v1.ConditionFalse,
+			Message: common.ImagePullFailureText,
+			Reason:  "ImagePullFailed",
+		}
+		utils.WaitForConditions(f, dv.Name, f.Namespace.Name, controllerSkipPVCCompleteTimeout, assertionPollInterval, runningCondition)
+
+	},
+		Entry("insecure registry", "registry:5000/myimage:latest"),
+		Entry("non exists image", "quay.io/capk/ubuntu-2004-container-disk:wrong"),
```

Comment:
Should not be a real issue, but maybe reference the internal registry here with an invalid image. This now assumes you have access to the internet, which may not always be valid.

Reply:
Any other idea?